app.js

```diff
@@ -20,13 +25,79 @@ liveReloadServer.server.once('connection', () => {
 })
 
 app.use('/static', express.static(publicDir))
+app.use(bodyParser.urlencoded({ extended: false }))
+app.set('view engine', 'pug')
+
+app.use(session({
+    secret: 'blablablaincrediblysafesecret',
+    resave: false,
+    saveUninitialized: false
+  }))
+
 
 app.get('/', (req, res) => {
-    res.sendFile(`${__dirname}/`)
+    res.render('')
+})
+
+app.get('/login', (req, res) => {
```

Comment:
Should we not make sure here that user is not logged in?
For example: 
```suggestion
app.get('/login', notLoggedInMiddlewareHere, (req, res) => {
```
Otherwise, even logged in users will have to log in again if, say, they access the '/login' url by accident.

And then maybe, if already logged in, just redirect them back to '/chat'.
Makes sense?

Reply:
Thank you for your time in reviewing this.

I was thinking on reusing the `isLoggedIn` middleware, but with an extra obj param that specifies what to check, like `{shouldBeLoggedIn: true}` (or `false`). Do you think that would suit this?

Reply:
It's not a bad idea.

However, it could bring some readability problems down the line. For example, since the middleware is called `isLoggedIn` it may not be immediately obvious when it's checking for the opposite.

Probably, if you create another middleware `isNotLoggedIn` or `IsLoggedOut`, intent will be more obvious.